benchmark/single-source/DataBenchmarks.swift

```diff
@@ -174,6 +191,12 @@ let small = sampleData(.small)
 let medium = sampleData(.medium)
 let large = sampleData(.large)
 
+let repeatElementSeq = { count in
+  return sequence(state: count) { (i: inout Int) -> UInt8? in
+    defer { i = i &- 1 }; return i > 0 ? UInt8(0xA0) : nil
```

Comment:
@itaiferber Could you please verify, that this closure is getting properly inlined? Just to be super sure the slowdown we're seeing is only caused by the byte-by-byte appending.

I guess this is the way to prove it? (haven't compiled, just typed in browser here...)
````swift
@inline(__always)
func nextOne(_ i: inout Int) -> UInt8? {
  defer { i = i &- 1 }; return i > 0 ? UInt8(0xA0) : nil
}

let repeatElementSeq = { count in
  return sequence(state: count, next: nextOne) 
}
````

Reply:
I'll try the above out locally, but I have a good feeling it is inlined because my experimental changes to chunk the results in `init` and `append` improve the benchmark by ~20x.

Reply:
That sounds great!